cmd/attest.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"google.golang.org/protobuf/encoding/prototext"
+	"github.com/google/go-tpm/tpm2"
+	"google.golang.org/protobuf/proto"	
+)
+var (
+	key      string
+	nonce    []byte
+	teenonce []byte
+	outputText string
+	gceZone  string
+    gceProjectID     string
+    gceProjectNumber uint64
+    gceInstanceName  string
+    gceInstanceID    uint64
+
+)
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func (rw io.ReadWriter)(*client.Key,error)  {
+	"AK":{
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gceAK":{
+		tpm2.AlgRSA:client.GceAttestationKeyRSA,
+		tpm2.AlgECC:client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest",
+	Short: "Create a remote attestation report",
+	Long: `Gather information for remote attestation.
+
+The Attestation report contains a quote on all available PCR banks ,a way to validate 
+the quote, and a TCG Event Log (Linux only).
+
+Use --key to specify the type of attestation key. It can be gceAK for GCE attestation
+key or AK for a custom attestation key.By default it uses gceAK.
+--algo flag override the public key algorithm for attestation key. If not provided then
+by default rsa is used.
+--nonce attaches even length extra data to the report and guarantees a fresh quote.
+--teenonce attaches a 64 bytes extra data to the attestation report of hardware and 
+guarantees a fresh quote.
+--output flag provides file name and location to store attestation report in the binary
+format. If filename is not provided then output will be printed on stdout.
+--outputText flag can be used to add text format of attestation report in a file.
```

Comment:
I will again register that you can download protoc and use it instead of building it into this CLI :).

Regardless, I think we should keep flags orthogonal as much as possible. Here, we could have a global flag that is useful elsewhere like `--format=prototext` and `--format=binarypb`. So instead of specifying `--outputText` and/or `--output` to print prototext, you just specify the format you want it in and `--output`.

Reply:
Noted. 
Instead of using --outputText , we will use --format to specify the type of output file.
But if format is set to prototext , then debugVerify will fail because it needs a binary file to read .

Reply:
then arguably, `--format` should be used for input as well in `debugVerify`